wp/plugin/lib/cache.ml

```diff
@@ -153,5 +152,20 @@ module Arch = struct
           KB.Conflict.pp c in
       failwith msg
 
+  (* Adds the program's architecture to the KB. *)
+  let save file arch : unit =
```

Comment:
It would be nice to add types here for consistency and readability

Reply:
I can definitely add them! It looks like I failed to add the types for the functions in this entire file, so I can add those too to be consistent.